### ISSUE TITLE
[in_app_purchase] Add currencySymbol

### ISSUE THREAD

**Issue by jimmyff** [flutter/flutter#83984]
It's great you've added `currencyCode` to the `ProductDetails` class, could you also add `currencySymbol`?

At the moment I have to do the following which feels pretty fragile:

```dart
final currencySymbol= productDetails.price.replaceAll(RegExp(r'[\d.,]'), '');
```

Thanks

**Comment by renefloor** [comment]
We are partly depending on the native SDKs to do this.
The iOS Storekit SDK has a priceLocale which does contain the currency code and symbol: https://developer.apple.com/documentation/storekit/skproduct/1506145-pricelocale

However, on Android the SkuDetails only contain the currency code: https://developer.android.com/reference/com/android/billingclient/api/SkuDetails

If we need to support this on both platforms we would have to manage all code -> currency.
I think it is is more suited to use a different package that is specialised for that conversion, such as [https://pub.dev/packages/country_currency_helper](country_currency_helper) (note that I think that package doesn't support parsing currency codes yet).

**Comment by stuartmorgan** [comment]
Is there a canonical way to do the complete formatting on Android? If so, and it's straightforward, we could potentially make a helper function that abstracts the process of creating the localized string, rather than exposing all the platform-specific components.

**Comment by renefloor** [comment]
@stuartmorgan the platform interface currently contains currencyCode (e.g. USD), rawPrice (e.g. 0.99) and price (e.g. $0.99). (See the [docs](https://pub.dev/documentation/in_app_purchase_platform_interface/latest/in_app_purchase_platform_interface/ProductDetails-class.html))

> Is there a canonical way to do the complete formatting on Android?

I think you mean exactly what 'price' returns right? @jimmyff is asking for a function that only returns '$' in this case, he does it by removing all numbers and ., from the price.

**Comment by stuartmorgan** [comment]
Oh, I see; I totally misunderstood the request.

> If we need to support this on both platforms we would have to manage all code -> currency.

Couldn't we just populate this from `Currency.getInstance(currencyCode).getSymbol()`?

**Comment by renefloor** [comment]
You mean doing that in Java? That makes sense.

**Comment by renefloor** [comment]
@stuartmorgan I created PR's for the android implementation and the platform_interface. The platform_interface is optional, but if we do that we should also update the app facing package and the ios package.

**Comment by mvanbeusekom** [comment]
The feature requested in this issue is implemented by PR flutter/plugins#4144 and as such I will close this issue now.
